packages/ui/components/input-file/src/LionInputFile.js

```diff
@@ -0,0 +1,879 @@
+        duplicateFileNamesValidator.param = { show: false };
+        this.validate();
+      }
+    }
+    return computedFileList;
+  }
+
+  /**
+   * @param {DragEvent} ev
+   */
+  uploadDroppedFiles(ev) {
+    ev.preventDefault();
+    this.removeAttribute('is-dragging');
+    if (ev.dataTransfer && ev.dataTransfer?.items.length > 1 && !this.multiple) {
+      return;
+    }
+
+    if (ev.dataTransfer?.files) {
+      this._inputNode.files = ev.dataTransfer.files;
+      // @ts-ignore
+      this.modelValue = Array.from(ev.dataTransfer.files);
+      // if same file is uploaded again, e.dataTransfer.files lists that file.
+      // So filter if the file already exists
+      // @ts-ignore
+      const _newFiles = this.computeNewAddedFiles(Array.from(ev.dataTransfer.files));
+      if (_newFiles.length > 0) {
+        this._uploadFiles(_newFiles);
+      }
+    }
+  }
+
+  /**
+   * @param {Event} ev
+   * @protected
+   */
+  // @ts-ignore
+  _onChange(ev) {
+    this.touched = true;
+    this._onUserInputChanged();
+    // @ts-ignore
+    this._uploadFiles(ev.target?.files);
+  }
+
+  /**
+   * Clear _inputNode.value to make sure onChange is called even for duplicate files
+   * @param {Event} ev
+   * @protected
+   */
+  // eslint-disable-next-line class-methods-use-this
+  _onClick(ev) {
+    // @ts-ignore
+    ev.target.value = ''; // eslint-disable-line no-param-reassign
+  }
+
+  /**
+   * @protected
+   */
+  _enhanceUploadedList() {
+    /**
+     * @type {LionUploadedFileList | null}
+     */
+    const uploadedFileList = this.querySelector('[slot="uploaded-file-list"]');
+    if (uploadedFileList) {
+      uploadedFileList.setAttribute('id', `uploaded-file-list-${this._inputId}`);
+      this.addToAriaDescribedBy(uploadedFileList, { idPrefix: 'uploaded-file-list' });
+    }
+  }
+
+  /**
+   * @protected
+   */
+  _syncAriaLabelledByAttributes() {
+    if (this._inputNode.hasAttribute('aria-labelledby')) {
+      const ariaLabelledBy = this._inputNode.getAttribute('aria-labelledby');
+      this._fileUploadButtonNode?.setAttribute(
+        'aria-labelledby',
+        `upload-button-${this._inputId} ${ariaLabelledBy}`,
+      );
+    }
+  }
+
+  /**
+   * @protected
+   */
+  _syncAriaDescribedByAttributes() {
+    if (this._inputNode.hasAttribute('aria-describedby')) {
+      const ariaDescribedby = this._inputNode.getAttribute('aria-describedby') || '';
+      this._fileUploadButtonNode?.setAttribute('aria-describedby', ariaDescribedby);
+    }
+  }
+
+  /**
+   * @private
+   */
+  __setupFileValidators() {
+    const { allowedFileTypes, allowedFileExtensions, maxFileSize } = this.fileLimit;
+
+    this.validators = [
+      ...this.validators,
+      new FileValidation({
+        allowedFileTypes,
+        allowedFileExtensions,
+        maxFileSize,
+      }),
+      new DuplicateFileNames({ show: false }),
+    ];
+  }
+
+  /**
+   * @param {InputFile[]} uploadedFiles
+   * @protected
+   */
+  _uploadFiles(uploadedFiles) {
```

Comment:
This method does not actually upload the files to a server and is therefore a bit misleading. Should we rename it to something like `processFiles` or `parseFiles`?